Lib/test/test_http_cookiejar.py

```diff
@@ -860,12 +860,24 @@ def test_is_HDN(self):
         self.assertTrue(is_HDN("foo.bar.com"))
         self.assertTrue(is_HDN("1foo2.3bar4.5com"))
         self.assertFalse(is_HDN("192.168.1.1"))
+        self.assertFalse(is_HDN("[::1]"))
+        self.assertFalse(is_HDN("[2001:db8:85a3::8a2e:370:7334]"))
         self.assertFalse(is_HDN(""))
         self.assertFalse(is_HDN("."))
         self.assertFalse(is_HDN(".foo.bar.com"))
         self.assertFalse(is_HDN("..foo"))
         self.assertFalse(is_HDN("foo."))
 
+    def test_is_ip(self):
+        self.assertTrue(is_ip('[::1]'))
```

Comment:
Strictly speaking, it's not an IPv6. It's a netloc with an IPv6 inside it. We should perhaps name `is_ip_like` instead and also treat `::1` as a valid IP (for blocked domains). I don't have an answer to that question so I'd prefer to know if there is some recommendations done by RFCs.